data-collection/data-collection/View Controllers/Floating Panel View Controller/FloatingPanelViewController.swift

```diff
@@ -0,0 +1,484 @@
+            
+            resizeableLayoutConstraint
+        ]
+        
+        // Define the constraints used for a compact-width layout.
+        compactWidthConstraints = [
+            view.leadingAnchor.constraint(equalTo: superview.safeAreaLayoutGuide.leadingAnchor, constant: compactWidthInsets.left),
+            view.trailingAnchor.constraint(equalTo: superview.safeAreaLayoutGuide.trailingAnchor, constant: compactWidthInsets.right),
+            view.bottomAnchor.constraint(equalTo: superview.bottomAnchor, constant: compactWidthInsets.bottom),
+            view.topAnchor.constraint(greaterThanOrEqualTo: superview.safeAreaLayoutGuide.topAnchor, constant: compactWidthInsets.top),
+            
+            headerStackView.bottomAnchor.constraint(lessThanOrEqualTo: superview.safeAreaLayoutGuide.bottomAnchor, constant: -compactWidthInsets.bottom),
+            
+            resizeableLayoutConstraint
+        ]
+        
+        // This will activate the appropriate set of constraints.
+        updateInterfaceForCurrentTraits()
+    }
+    
+    override public func viewDidAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        // Skip if already initialized or we don't have a superview.
+        guard !initialized else { return }
+        setupConstraints()
+        initialized = true
+    }
+    
+    /// Update constraints for current horizontal size class.
+    private func updateInterfaceForCurrentTraits() {
+        NSLayoutConstraint.deactivate(regularWidthConstraints)
+        NSLayoutConstraint.deactivate(compactWidthConstraints);
+        
+        NSLayoutConstraint.activate(isCompactWidth ? compactWidthConstraints : regularWidthConstraints);
+    }
+    
+    /// Handles a user tap on the close button
+    /// - Parameter sender: The button tapped.
+    @IBAction private func closeButtonAction(_ sender: Any) {
+        delegate?.userDidRequestDismissFloatingPanel(self)
+    }
+    
+    /// Handles the pan gesture used to resize the floating panel.
+    /// - Parameter gestureRecognizer: The active gesture recognizer.
+    @IBAction private func handlePanGesture(_ gestureRecognizer : UIPanGestureRecognizer) {
+        // Get the changes in the X and Y directions relative to
+        // the superview's coordinate space.
+        let translation = gestureRecognizer.translation(in: gestureRecognizer.view?.superview)
+        
+        if gestureRecognizer.state == .began {
+            // Save the view's original constant.
+            initialResizableLayoutConstraintConstant = resizeableLayoutConstraint.constant
+        }
+        
+        // Velocity will be used to determine whether to handle
+        // the 'flick' gesture to switch between states.
+        let velocity = gestureRecognizer.velocity(in: gestureRecognizer.view?.superview)
+        if gestureRecognizer.state == .ended {
+            if abs(velocity.y) > 0 {
```

Comment:
In most cases, instance methods are preferred over global functions.
```suggestion
            if velocity.y.magnitude > 0 {
```